pipelines/batch/src/main/java/com/google/fhir/analytics/ParquetMerger.java

```diff
@@ -82,13 +91,18 @@ private static PCollection<KV<String, Iterable<GenericRecord>>> readAndGroupById
             .apply(FileIO.matchAll())
             .apply(FileIO.readMatches());
 
-    // The assumption here is that the schema of the old parquet files is same as the current
-    // schema, otherwise reading of the older records fail even if the new Schema is just an
-    // extension. In general, if the schema changes due to addition of new extensions, then the
-    // merging process fail. It is recommended to recreate the entire parquet files using the new
-    // schema again using the batch run.
-    PCollection<GenericRecord> records =
-        inputFiles.apply(ParquetIO.readFiles(avroConversionUtil.getResourceSchema(resourceType)));
+    PCollection<GenericRecord> records;
+    if (viewDefinition != null) {
```

Comment:
Consider passing the Schema as an argument to the method which should be resolved by the caller, so that we don't have this if-else logic over here.

Reply:
Good catch this simplifies the logic nicely, the changes will be reflected in a separate Incremental Run PR.